route/route.go

```diff
@@ -51,7 +54,24 @@ func (r *Route) addTarget(service string, targetURL *url.URL, fixedWeight float6
 	name := metrics.TargetName(service, r.Host, r.Path, targetURL)
 	timer := gometrics.GetOrRegisterTimer(name, metrics.ServiceRegistry)
 
-	t := &Target{Service: service, Tags: tags, URL: targetURL, FixedWeight: fixedWeight, Timer: timer, timerName: name}
+	stripPrefix := false
+	// by passing a tag of `url-prefix-host/modifier#strip-prefix you can
+	// drop whatever prefix the service is namespaced by
+	for _, tag := range tags {
+		if StripPrefixRegex.MatchString(tag) {
+			stripPrefix = true
+		}
+	}
+
+	t := &Target{
```

Comment:
What do you think about a `NewTarget` method, which accepts the raw parameters and makes some decisions based upon which `tags` are specified?